### PR TITLE
chore(ui-react-core): improve internal documentation of useHasValueUpdated

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add internal documentation for `useHasValueUpdated` and swap transposed unit test descriptions

image

#### Issue #, if available

NA

#### Description of how you validated changes

NA

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-core/src/hooks/__tests__/useHasValueUpdated.spec.ts [hunk 1]
```diff
@@ -3,14 +3,14 @@ import { renderHook } from '@testing-library/react-hooks';
 import useHasValueUpdated from '../useHasValueUpdated';
 
 describe('useHasValueUpdated', () => {
-  it('return true on initial render and ignoring first render', () => {
+  it('return true on initial render and not ignoring first render', () => {
```
Comment (zchenwei) [inline packages/react-core/src/hooks/__tests__/useHasValueUpdated.spec.ts:6]: ⭐

**Review by zchenwei — APPROVED** [review]
LGTM. Thanks for adding the documentation!

File: packages/react-core/src/hooks/useHasValueUpdated.ts [hunk 2]
```diff
@@ -2,6 +2,19 @@ import { isUndefined } from '@aws-amplify/ui';
...
+ * @param value `value` to track for updates
+ * @param ignoreFirstRender whether to ignore initial render. defaults to `false`
+ * @returns a boolean representing whether the tracked `value` has updated between renders
+ *
+ * Returns `false`:
+ * - on initial render when ignoring first render
+ * - current and previous `value` are equal
+ *
+ * Returns `true`:
+ * - on initial render and not ignoring first render (default behavior)
```
Comment (ioanabrooks) [inline packages/react-core/src/hooks/useHasValueUpdated.ts:15]:
```suggestion
 * - on initial render when not ignoring first render (default behavior)
```
?

Comment (calebpollman) [reply packages/react-core/src/hooks/useHasValueUpdated.ts:15]: What's the question 😅

**Review by zchenwei — APPROVED** [review]

**Review by dbanksdesign — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**